File: Ruby/day1_part2.rb
depth = File.open("sonar.txt", chomp: true).map(&:to_i)

def three_pair_depth_counter(depth)
    count=0
    sum=adjacent_sum(depth)
    (0...sum.length-1).each do |i|
        if sum[i+1]>sum[i]
            count+=1
        end
    end
    count
end

def adjacent_sum(arr)
    sum=[]
    (0...arr.length-2).each do |i|
        first = arr[i]
        sec = arr[i+1]
        third = arr[i+2]
        sum<<(first+sec+third)
    end
    sum
end

p three_pair_depth_counter(depth)

File: Python/day1_part2.py
with open("sonar.txt") as f:
    lines=f.read()
    depth=lines.split("\n")
    
def depth_sum_counter(depth):
    adj_sum=adjacent_sum(depth)
    count=0
    for i in range(len(adj_sum)-1):
        if adj_sum[i+1]>adj_sum[i]:
            count+=1
    return(count)

def adjacent_sum(arr):
    sum=[]
    for i in range(len(arr)-2):
        first=int(arr[i])
        second=int(arr[i+1])
        third=int(arr[i+2])
        equals=first+second+third
        sum.append(equals)
    return(sum)

print(depth_sum_counter(depth))
